Training/Program.cs

```diff
@@ -0,0 +1,106 @@
+      }
+      #endregion
+   }
+   #endregion
+
+   #region Class Double ended TQueue --------------------------------------------------------------
+   /// <summary>Double Ended Queue<T></summary>
+   /// <typeparam name="T">Type of the Queue</typeparam>
+   public class TDQueue<T> {
+      #region Constructor -------------------------------------------
+      public TDQueue () => mElements = new T[mSize];
+      #endregion
+      #region Property ----------------------------------------------
+      /// <summary>Returns true if the Queue is empty otherwise false</summary>
+      public bool IsEmpty => mCount == 0;
+      #endregion
+
+      #region Methods -----------------------------------------------
+      /// <summary>Adds the element at the rear end of the Queue</summary>
+      /// <param name="a">Element to be added to the Queue</param>
+      public void RearEnqueue (T a) {
+         if (mCount == mSize) ResizeArray ();
+         mElements[rear] = a;
+         rear = (rear + 1) % mSize;
+         mCount++;
+      }
+
+      /// <summary>Returns and removes the element at the rear end of the Queue</summary>
+      /// <returns>The element at the rear end of the Queue</returns>
+      /// <exception cref="InvalidOperationException">Exception thrown if the queue is empty</exception>
+      public T RearDequeue () {
+         if (IsEmpty) throw new InvalidOperationException ();
+         rear = (rear - 1 + mSize) % mSize;
+         T a = mElements[rear];
+         mCount--;
+         return a;
+      }
+
+      /// <summary>Adds the element at the front end of the Queue</summary>
+      /// <param name="a">Element to be added to the Queue</param>
+      public void FrontEnqueue (T a) {
+         if (mCount == mSize) ResizeArray ();
+         front = (front -1 + mSize) % mSize;
+         mElements[front] = a;
+         mCount++;
+      }
+
+      /// <summary>Returns and removes the element at the front end of the Queue</summary>
+      /// <returns>The element at the front end of the Queue</returns>
+      /// <exception cref="InvalidOperationException">Exception thrown if the queue is empty</exception>
+      public T FrontDequeue () {
+         if (IsEmpty) throw new InvalidOperationException ();
+         T a = mElements[front];
+         front = (front + 1) % mSize;
+         mCount--;
+         return a;
+      }
+
+      /// <summary>Expands the length of the array</summary>
+      void ResizeArray () {
+         T[] expArray = new T[mCount * 2];
+         rear--;
+         for (int i = front, j = 0; i >= rear; i++, j++) {
+            int ind = i % mSize;
+            expArray[j] = mElements[ind];
+            if (ind == rear) break;
+         }
+         mElements = expArray;
+         front = 0;
+         rear = mCount;
+         mSize *= 2;
+      }
+      #endregion
+
+      #region Private Fields ----------------------------------------
+      int front = 0, rear = 0, mCount = 0, mSize = 4;

```

Comment:
```suggestion
      int mFront = 0, mRear = 0, mCount = 0, mSize = 4;

```
Member variables should start with m